smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/TaggableApiConfig.java

```diff
@@ -0,0 +1,157 @@
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.traits.tagging;
+
+import software.amazon.smithy.model.FromSourceLocation;
+import software.amazon.smithy.model.SourceLocation;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NodeMapper;
+import software.amazon.smithy.model.node.ToNode;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.utils.SmithyBuilder;
+import software.amazon.smithy.utils.ToSmithyBuilder;
+
+/**
+ * Structure representing the configuration of resource specific tagging APIs.
+ */
+public final class TaggableApiConfig implements FromSourceLocation, ToNode, ToSmithyBuilder<TaggableApiConfig> {
+    private final ShapeId tagApi;
+    private final ShapeId untagApi;
+    private final ShapeId listTagsApi;
+    private final SourceLocation sourceLocation;
+
+    private TaggableApiConfig(Builder builder) {
+        tagApi = builder.tagApi;
+        untagApi = builder.untagApi;
+        listTagsApi = builder.listTagsApi;
+        sourceLocation = builder.sourceLocation != null
+                ? builder.sourceLocation
+                : SourceLocation.none();
+    }
+
+    /**
+     * Creates a builder used to build an TaggableApiConfig.
+     *
+     * @return Returns the created builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Gets the ShapeId of the operation which implements TagResource behavior.
+     *
+     * @return the ShapeId of the tag operation for the resource
+     */
+    public ShapeId getTagApi() {
+        return tagApi;
+    }
+
+    /**
+     * Gets the ShapeId of the operation which implements UntagResource behavior.
+     *
+     * @return the ShapeId of the untag operation for the resource
+     */
+    public ShapeId getUntagApi() {
+        return untagApi;
+    }
+
+    /**
+     * Gets the ShapeId of the operation which implements ListTagsForResource
+     * behavior for the resource.
+     *
+     * @return the ShapeId of the list tags operation for the resource
+     */
+    public ShapeId getListTagsApi() {
+        return listTagsApi;
+    }
+
+    @Override
+    public SourceLocation getSourceLocation() {
+        return sourceLocation;
+    }
+
+    @Override
+    public Builder toBuilder() {
+        Builder builder = builder();
+        return builder
+                .tagApi(tagApi)
+                .untagApi(untagApi)
+                .listTagsApi(listTagsApi)
+                .sourceLocation(sourceLocation);
+    }
+
+    @Override
+    public Node toNode() {
```

Comment:
this toNode method is used in .equals and it's kind of expensive due to the NodeMapper. Seems easy enough to use a Node.objectNodeBuilder